patches/minecraft/net/minecraft/block/BlockRedstoneDiode.java.patch

```diff
@@ -0,0 +1,29 @@
+--- ../src-base/minecraft/net/minecraft/block/BlockRedstoneDiode.java
++++ ../src-work/minecraft/net/minecraft/block/BlockRedstoneDiode.java
+@@ -1,6 +1,7 @@
+ package net.minecraft.block;
+ 
+ import java.util.Random;
++
+ import net.minecraft.block.material.Material;
+ import net.minecraft.block.state.IBlockState;
+ import net.minecraft.entity.EntityLivingBase;
+@@ -48,7 +49,7 @@
+         if (!this.func_176405_b(p_180650_1_, p_180650_2_, p_180650_3_))
+         {
+             boolean flag = this.func_176404_e(p_180650_1_, p_180650_2_, p_180650_3_);
+-
++            
+             if (this.field_149914_a && !flag)
+             {
+                 p_180650_1_.func_180501_a(p_180650_2_, this.func_180675_k(p_180650_3_), 2);
+@@ -113,6 +114,9 @@
+         {
+             boolean flag = this.func_176404_e(p_176398_1_, p_176398_2_, p_176398_3_);
+ 
++            net.minecraftforge.event.world.RedstoneUpdateEvent event = net.minecraftforge.event.ForgeEventFactory.onRedstoneUpdate(p_176398_1_, p_176398_2_, p_176398_3_, field_149914_a ? 15 : 0, flag ? 15 : 0);
```

Comment:
Same here, it's called both client and server, thus causing potential desyncs and you have no protection against this.